autoscaler/controllers/datacollection/daemonset.go

```diff
@@ -38,17 +37,43 @@ var (
 	}
 )
 
-func getOdigletDaemonsetPodSpec(ctx context.Context, c client.Client, namespace string) (*corev1.PodSpec, error) {
-	odigletDaemonset := &appsv1.DaemonSet{}
+type DelayManager struct {
+	mu         sync.Mutex
+	inProgress bool
+}
 
-	if err := c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: odigletDaemonSetName}, odigletDaemonset); err != nil {
-		return nil, err
+// RunSyncDaemonSetWithDelayAndSkipNewCalls runs the function with the specified delay and skips new calls until the function execution is finished
+func (dm *DelayManager) RunSyncDaemonSetWithDelayAndSkipNewCalls(delay time.Duration, retries int, dests *odigosv1.DestinationList, collection *odigosv1.CollectorsGroup, ctx context.Context, c client.Client, scheme *runtime.Scheme, secrets []string, version string) {
+	dm.mu.Lock()
+	defer dm.mu.Unlock()
+
+	// Skip new calls if the function is already in progress
+	if dm.inProgress {
+		return
 	}
 
-	return &odigletDaemonset.Spec.Template.Spec, nil
+	dm.inProgress = true
+
+	// Finish the function execution after the delay
+	time.AfterFunc(delay, func() {
+		dm.mu.Lock()
+		defer dm.mu.Unlock()
+		defer dm.finishProgress()
+
+		for i := 0; i < retries; i++ {
+			_, err := syncDaemonSet(ctx, dests, collection, c, scheme, secrets, version)
+			if err == nil {
+				return
+			}
+		}
```

Comment:
should we log something if we run out of retries and give up?
otherwise, the error is silently dropped, right?

and should we also log something on success to aid debugging in case there are issues?